### ISSUE TITLE
Multiple content in one target + transition

### ISSUE THREAD

**Issue by andrey-hohlov** [LinusBorg/portal-vue#141]
Hi!

I'm not sure if this bug, or maybe I'm doing something wrong.
If the second - let's document it for peoples like me)

We have multiple components - dropdown menus, tooltips and other.
Let's name them 'popups'. 
They are different but all of them we need to display in the root of our App to avoid positioning troubles.

We created the target for all of these components:

```vue
<portal-target name="app-popups" multiple/>
```

Each component usually has template like this:

```vue
<div>
  <button
    type="button"
    @click="show = !show"
  >
    Activator button
  </button>
  <portal to="app-popups">
    <transition name="component-specific-transition">
      <div v-if="show">Content</div>
    </transition>
  </portal>
</div>
```

It works at first sight.
But if we have many components - troubles coming.

I recorded video to show: http://take.ms/6FENv

Loop with v-for, or just many duplications in the templates - doesn't matter.
Wrapping transition in additional div - doesn't help.
Just remove transition from component - and all works well.

Demo: https://andrey-hohlov.github.io/demo-portal-vue-multiple-content-transition-bug/
Repo: https://github.com/andrey-hohlov/demo-portal-vue-multiple-content-transition-bug

For my case I solved it using transition on `portal-target`. This's okay and this allow to customize transition in components. But if I understood correctly this limits the use of functions for transition.

**Comment by LinusBorg** [comment]
So I looked at the demo (video is not working for me right now), and I'm not sure what I should be looking for, what the bad behaviour is supposed to be.

The demo allows me to fade in and fade out different elements, and for me, they all fade in and out nicely. 

Can you actually write down what the problem is, please? That would be helpful.

**Comment by andrey-hohlov** [comment]
@LinusBorg sorry, easier to show then write)

Looks like a connection between the component and its teleported part breaks down.

Just click all buttons, from first to last to fade in all teleported elements.
Then click all buttons again, to fade out all elements.
You will see that all components have `show: false` (text on the buttons is 'Show'), but 2 teleported elements still visible (not sure that it always 2).

Button and content render component uid. In my case components with uids 19 and 21 are broken. I click on the button, `show` changes, text on the button changes. But teleported content - doesn't. It still displays in portal-target.

I hope it's enough to explain.

**Comment by LinusBorg** [comment]
Hm, I just clicked all buttons top to bottom, to sho them, and then again to hide them. All components are gone. So I can't really reproduce the behaviour you describe.

Generally, I would strongly recommend to `key` the content of the transitions, that might fix whatever you see.

**Comment by andrey-hohlov** [comment]
Strange...

Video on YouTube: https://youtu.be/XBuWKykg4P0
Chrome 67.0.3396.99, Mac OS 10.13.5

I didn't expect that it can depend on the browser. But I tried Firefox and Safari - and it works well.

Chrome 68.0.3440.84, Windows 7 (clean install in VirtualBox) - the same bug!
I asked my friend (Windows 10 + Chrome 67) - the same bug, 19 and 21 always freeze.

I added key

```
data () {
  return {
    show: false,
    pid: `component-${Date.now()}`,
  };
},
```

```vue
<portal to="destination-multiple">
  <transition name="fade">
    <div
      v-if="show"
      :key="pid"
   >
     Hi (uid {{ _uid }})!
   </div>
 </transition>
</portal>
```
but it still not works.

**Comment by LinusBorg** [comment]
Thanks for the video- that looks even weirder than I thought. Fo me, items always rendered in the right order, and all items disappeared again like they were supposed to.

Seems to be browser-related indeed. Those bugs are just the worst ^^

About the `key`: did you verify that there were actually no duplicate keys? I kow Date.now() returns milliseconds, but rendering a few elements in a loop like this is *really* fast on a good machine .

Also: These lines could be problematic:

https://github.com/andrey-hohlov/demo-portal-vue-multiple-content-transition-bug/blob/master/src/App.vue#L3-L6

You essentially use an index as a key, which means when you remove components in the middle of the list, components switch keys etc - exactly not what we want.

**Comment by andrey-hohlov** [comment]
I pushed an update with changes to minimize potential bug-factors.

1. Removed for-in loop (just duplicate components in the template).
2. Added more random numbers for the key. I think it is highly unlikely that they will be duplicated. You were right, Date.now() is not enough.

Bug still exists)

**Comment by tmorehouse** [comment]
@andrey-hohlov could you try the latest version (1.5.0) and see if the introduction of `stableSort` in `worrmhole` makes a difference?

**Comment by andrey-hohlov** [comment]
@tmorehouse there is a problem - now I have Chrome 71.0.3578.98 and can't reproduce issue :(

**Comment by LinusBorg** [comment]
That may actually be a good sign since the problem addressed by the fix for stable sorting  in 1.5.0 was about Chrome, mostly. Firefox and Safari already did a stable sort without any "help".

So that may mean the fix really solved your issue.

**Comment by andrey-hohlov** [comment]
@LinusBorg I mean I checked 1.3 version demo at first https://andrey-hohlov.github.io/demo-portal-vue-multiple-content-transition-bug/ but with Chrome 71 and it works correctly.
1.5 works on Chrome 71 correctly too.

Fortunately, just now I found VM with Windows and Chrome 68, checked 1.3 - bug exist.
Checked 1.5 - and bug fixed!

So, resume:
- Bug fixed in Chrome (71+, maybe 69+)
- Bug fixed in portal-vue 1.5

Thanks for all!
